lib/irb/completion.rb

```diff
@@ -116,18 +139,27 @@ def completion_candidates(preposing, target, _postposing, bind:)
       # When completing the argument of `help` command, only commands should be candidates
       return command_candidates(target) if preposing.match?(HELP_COMMAND_PREPOSING)
 
-      commands = if preposing.empty?
-        command_candidates(target)
+      type_candidates = type_completion_candidates(preposing, target, bind)
+
+      if preposing.empty?
+        command_candidates(target) | type_candidates
       # It doesn't make sense to propose commands with other preposing
       else
-        []
+        type_candidates
       end
+    end
 
+    def type_completion_candidates(preposing, target, bind)
       result = ReplTypeCompletor.analyze(preposing + target, binding: bind, filename: @context.irb_path)
+      return [] unless result
 
-      return commands unless result
-
-      commands | result.completion_candidates.map { target + _1 }
+      analyze_result = result.instance_variable_get(:@analyze_result)
+      if analyze_result.is_a?(Array) && analyze_result[0] == :symbol && analyze_result[1].is_a?(String)
```

Comment:
I think it is relatively safe because it checks the value strictly, but it's a bit hacky...